docs/source/release_notes/release-notes.rst

```diff
@@ -67,9 +69,9 @@ develop
            (`Pull Request <https://github.com/palantir/atlasdb/pull/1265>`__)
 
     *    - |devbreak|
```

Comment:
I would remove the last sentence about not properly deprecating. If we are confident that this API isn't used, then let's say so. If we aren't sure, then let's revert the break, deprecate it, and remove it in a release or two.

My guess is we are very confident it's not used by anybody and it was just broken before, so there's little risk to removing it.